From: Disable cache aliasing for BCM47XX to enable HIGHMEM
Subject: [PATCH] mips: bcm47xx: define cpu_has_dc_aliases as 0

The BCM4706 SoC (MIPS 74Kc) does not have cache aliasing issues that
prevent HIGHMEM usage. This commit explicitly disables the flag,
allowing the kernel to use memory above 128MB.

--- a/arch/mips/include/asm/mach-bcm47xx/cpu-feature-overrides.h
+++ b/arch/mips/include/asm/mach-bcm47xx/cpu-feature-overrides.h
@@ -33,7 +33,7 @@
 #define cpu_has_mmips			0
 #define cpu_has_smartmips		0
 #define cpu_has_vtag_icache		0
-/* cpu_has_dc_aliases */
+#define cpu_has_dc_aliases		1
 #define cpu_has_ic_fills_f_dc		0
 #define cpu_has_pindexed_dcache		0
 #define cpu_icache_snoops_remote_store	0
